### PR TITLE
fix: Invalid merge inclusion

### PR DESCRIPTION
**Pull request by jeromelaban**

### CODE REVIEW

File: src/Uno.XamlMerge.Task/FullPathComparer.cs [hunk 1]
```diff
@@ -10,8 +10,7 @@ internal class FullPathComparer : IEqualityComparer<ITaskItem>
 
     public bool Equals(ITaskItem x, ITaskItem y)
     {
-        return x.GetMetadata("FullPath") == y.GetMetadata("FullPath")
-           && x.GetMetadata("MergeFile") == y.GetMetadata("MergeFile");
+        return x.GetMetadata("FullPath") == y.GetMetadata("FullPath");
```
Comment (kazo0) [inline src/Uno.XamlMerge.Task/FullPathComparer.cs:13]: We don't need the `GetMetadata("MergeFile")` comparison any more?

Comment (jeromelaban) [reply src/Uno.XamlMerge.Task/FullPathComparer.cs:13]: It's not working in the previous PR, so no :) The problem is the comparison at the entry of the Task, for deduplication.

File: src/Uno.XamlMerge.Task/BatchMergeXaml.cs [hunk 2]
```diff
@@ -40,9 +40,8 @@ public override bool Execute()
                 return false;
             }
 
-            var filteredPages = Pages
-                .Except(MergedXamlFiles, FullPathComparer.Default)
-                .ToArray();
+            var filteredPages = Pages.ToList();
+            filteredPages.RemoveAll(e => MergedXamlFiles.Any(m => FullPathComparer.Default.Equals(e, m));
```
Comment (kazo0) [inline src/Uno.XamlMerge.Task/BatchMergeXaml.cs:44]:
```suggestion
            filteredPages.RemoveAll(e => MergedXamlFiles.Any(m => FullPathComparer.Default.Equals(e, m)));
```

Comment (jeromelaban) [reply src/Uno.XamlMerge.Task/BatchMergeXaml.cs]: w00ps ;)

**Review by kazo0 — APPROVED** [review]

**Review by kazo0 — APPROVED** [review]

**Pull request merged**